src/Bicep.LangServer/Handlers/BicepDeploymentStartCommandHandler.cs

```diff
@@ -63,6 +78,36 @@ public override async Task<BicepDeploymentStartResponse> Handle(BicepDeploymentS
             var credential = new CredentialFromTokenAndTimeStamp(request.token, request.expiresOnTimestamp);
             var armClient = new ArmClient(credential, default, options);
 
+            string? bicepparamJsonOutput = null;
+
+            if(request.parametersFilePath.EndsWith(".bicepparam"))
+            {
+                //params file validation 
+                if(request.parametersFileUpdateOption != ParametersFileUpdateOption.None)
+                {
+                    return new BicepDeploymentStartResponse(false, "Can not create/overwrite/update parameter files when using a bicep parameters file", null);
+                }
+
+                if(request.updatedDeploymentParameters.Any())
+                {
+                    return new BicepDeploymentStartResponse(false, "Can not update parameters for bicep parameter file", null);
+                }
+
+                //params file compilation
+                var documentUri = DocumentUri.FromFileSystemPath(request.parametersFilePath);
+                var compilation = await new CompilationHelper(bicepCompiler, compilationManager).GetCompilation(documentUri);
+                
+                var diagnosticsByFile = compilation.GetAllDiagnosticsByBicepFile()
+                .FirstOrDefault(x => x.Key.FileUri == documentUri.ToUri());
+                
+                if (diagnosticsByFile.Value.Any(x => x.Level == DiagnosticLevel.Error))
+                {
+                   return new BicepDeploymentStartResponse(false, "Bicep build failed. Please fix below errors:\n" + DiagnosticsHelper.GetDiagnosticsMessage(diagnosticsByFile), null);
+                };
+                
+                bicepparamJsonOutput = GetParamsJsonOutput(compilation);
+            } 
```

Comment:
Could this whole `if` block be factored out into a separate method?

Reply:
The compilation logic can moved into a separate method but validation on `parametersFileUpdateOption` and `updatedDeploymentParameters` need to be here I think